tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -267,20 +809,29 @@ function Use-AzureRmProfile
     $ProfileMap = (Get-AzProfile)
     $AllProfiles = ($ProfileMap | Get-Member -MemberType NoteProperty).Name
     $Profile = $PSBoundParameters.Profile
+    $Scope = $PSBoundParameters.Scope
+    $Modules = $PSBoundParameters.Module
+
+    if ($Modules -eq $null)
+    {
+      $Modules = ($ProfileMap.$Profile | Get-Member -MemberType NoteProperty).Name
+    }
+
     if ($PSCmdlet.ShouldProcess($Profile, "Loading modules for profile in the current scope"))
```

Comment:
I would remove this SHouldProcess in favor of one for each module we install and each module we import